lib/services/dynamodb.js

```diff
@@ -47,7 +47,18 @@ AWS.util.update(AWS.DynamoDB.prototype, {
    * @api private
    */
   retryDelays: function retryDelays(retryCount) {
-    var delay = retryCount > 0 ? (50 * Math.pow(2, retryCount - 1)) : 0;
+    var retryDelayOptions = this.config.retryDelayOptions;
+
+    if (typeof retryDelayOptions.base !== 'number') {
+        retryDelayOptions.base = 50; // default for dynamodb
+    }
+
+    var delay;
+    if (typeof retryDelayOptions.customBackoff === 'function') {
+      delay = AWS.util.calculateRetryDelay(retryCount, retryDelayOptions);
+    } else {
+      delay = retryCount > 0 ? AWS.util.calculateRetryDelay(Math.max(0, retryCount - 1), retryDelayOptions) : 0;
```

Comment:
I'm not sure this is in line with the recommendation. [DynamoDB recommends](http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Programming.Errors.html#Programming.Errors.RetryAndBackoff) that the first retry be delayed up to 50 ms, the second up to 100 ms, etc.; whereas this will follow that sequence for retries 2-10 with the first retry being dispatched with no delay.

Applying the same delay calculation to all retries also means that you can just return `AWS.util.calculateRetryDelay(retryCount, retryDelayOptions)` instead of having extra branching logic.

See https://github.com/aws/aws-sdk-php/issues/1186 for discussion on this in another SDK.